### ISSUE TITLE
Cargo reports unknown lint clippy::if_then_some_else_none

### ISSUE THREAD

**Issue by TheDutchMC** [rust-lang/rust-clippy#7337]
I tried this code:
````rs
#![warn(clippy::if_then_some_else_none)]
````

I expected to see this happen: *explanation*
Cargo/Clippy to accept this and warn me of any cases where this list applies. The lint is mentioned (here)[https://rust-lang.github.io/rust-clippy/master/index.html]

Instead, this happened: *explanation*
Cargo/Clippy doesn't recognize the lint

### Meta

- `cargo clippy -V`: clippy 0.1.52 (9bc8c42 2021-05-09)
- `rustc -Vv`:
```
rustc 1.52.1 (9bc8c42bb 2021-05-09)
binary: rustc
commit-hash: 9bc8c42bb2f19e745a63f3445f1ac248fb015e53
commit-date: 2021-05-09
host: x86_64-unknown-linux-gnu
release: 1.52.1
LLVM version: 12.0.0
```

**Comment by xFrednet** [comment]
The link you looked at is the documentation for our current master state. The `if_then_some_else_none` lint will be introduced in [Rust 1.53](https://github.com/rust-lang/rust-clippy/blob/master/CHANGELOG.md#rust-153) released on 2021-06-17. All lints in stable Clippy can be found here:

https://rust-lang.github.io/rust-clippy/stable/index.html

**Comment by TheDutchMC** [comment]
Ah that explains it, didn't notice that.

Thanks!
